TEST.md

```diff
@@ -0,0 +1,192 @@
+# Conduit Test Guide
+
+This document covers how to run all of the tests that are present in the Conduit
+repo. Most of these tests are run in CI, but you can use the instructions here
+to run the tests from source. For more information about working in this repo,
+see the [BUILD.md](BUILD.md) guide.
+
+Note that all bash commands in this guide are expected to be run from the root
+of this repo, unless otherwise indicated by a `cd` command.
+
+# Table of contents
+
+- [Integration tests](#integration-tests)
+  - [Prerequisites](#prerequisites)
+  - [Running tests](#running-tests)
+  - [Cleanup](#cleanup)
+  - [Writing tests](#writing-tests)
+- [Integration tests: proxy-init](#integration-tests-proxy-init)
+- [Unit tests](#unit-tests)
+  - [Rust](#rust)
+  - [Go](#go)
+  - [React](#react)
+
+# Integration tests
+
+The `test/` directory contains a test suite that can be run to validate Conduit
+functionality via a series of end-to-end tests.
+
+## Prerequisites
```

Comment:
consider just calling out that this works in minikube and gke, and doesn't work in docker for mac + k8s.